packages/mobile/src/store/settings/sagas.ts

```diff
@@ -18,13 +18,51 @@ import PushNotifications from 'app/notifications'
 const { getPushNotificationSettings } = settingsPageSelectors
 const { getAccountUser } = accountSelectors
 
-export function* disablePushNotifications() {
+export function* deregisterPushNotifications() {
   const audiusBackendInstance = yield* getContext('audiusBackendInstance')
   const { token } = yield* call([PushNotifications, 'getToken'])
   PushNotifications.deregister()
   yield* call(audiusBackendInstance.deregisterDeviceToken, token)
 }
 
+function* enablePushNotifications() {
+  yield* call([PushNotifications, 'requestPermission'])
+  const { token, os } = yield* call([PushNotifications, 'getToken'])
+
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+
+  // Enabling push notifications should enable all of the notification types
+  const newSettings = { ...initialState.pushNotifications }
+  yield* put(actions.setPushNotificationSettings(newSettings))
+
+  // We need a user for this to work (and in the case of sign up, we might not
+  // have one right away when this function is called)
+  yield* call(waitForValue, getAccountUser)
+  yield* call(audiusBackendInstance.updatePushNotificationSettings, newSettings)
+
+  yield* call(audiusBackendInstance.registerDeviceToken, token, os)
+}
+
+function* disablePushNotifications() {
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const newSettings = { ...initialState.pushNotifications }
+  for (const key in newSettings) {
+    newSettings[key] = false
+  }
```

Comment:
We could use lodash `mapValues` for this!

```suggestion
  const newSettings = mapValues(initialState.pushNotifications, false)
```

Reply:
amazing, will add